src/coreclr/jit/emitxarch.cpp

```diff
@@ -3848,6 +3848,35 @@ inline emitter::insFormat emitter::emitInsModeFormat(instruction ins, insFormat
     return (insFormat)(base + emitInsUpdateMode(ins));
 }
 
+/*****************************************************************************
+ *
+ *  Combine the given base format with the update mode of the APX-NDD instruction.
+ */
+
+inline emitter::insFormat emitter::emitNddInsModeFormat(instruction ins, insFormat base)
+{
+#ifdef TARGET_AMD64
+    assert(IsApxNDDEncodableInstruction(ins));
+    switch (base)
```

Comment:
You're handling a few formats here, but I only see it being called with `IF_RRD_RRD_ARD`. It looks like `IF_RRD_RRD_MRD` is also missing here.

Are there other places using the `useNDD ? emitNddInsModeFormat(ins, IF_RRD_RRD_*RD) : emitInsModeFormat(ins, IF_RRD_*RD);` pattern?